### ISSUE TITLE
MAS 4.1.2: Ensures buttons have discernible text (a[role="button"]:nth-child(4))

### ISSUE THREAD

**Issue by azaslonov** [paperbits/paperbits-demo#137]
User Experience:

User would find difficulty to identify the controls if name and title are not defined for any control as they won't get the context of that control and also they won't be able to perform functionality  relevant to that control.

 

Note: User credentials should NOT be included in the bug.

Repro-steps:

1. Open Azure portal with valid credentials and create a API Management.

2. Click Developer Portal button to navigate to the portal.

3. Wait portal to load in design mode

4. Navigate to the Left nav and select Layouts button.

5. Navigate to the "Layouts" window and select Home option.

6. Navigate to the "Aside " option and select edit button.

7. Navigate to the various controls of the "Aside "dialog box like content alignment, content flow.

8. Run ‘Accessibility insights for web’ tool and select Fast Pass options.

9. Verify the issue.

 

Actual Result:

Name is not defined for horizontal and vertical controls present in ‘Aside’ popup.

 

Expected Result:

Name and title should be defined for horizontal and vertical controls present in ‘Aside’ popup.

 

Issue: Ensures buttons have discernible text (button-name - https://dequeuniversity.com/rules/axe/3.3/button-name?application=msftAI)

 

Target application: API Management - https://accessibility.developer.azure-api.net/apis

 

Element path: .form-group:nth-child(2) > a[role="button"]:nth-child(4)

 

Snippet: <a href="#" class="toolbox-btn" role="button"> <i class="paperbits-icon paperbits-align-bottom"></i> </a>

 

How to fix:

Fix any of the following:

  Element does not have inner text that is visible to screen readers

  aria-label attribute does not exist or is empty

  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty

  Element's default semantics were not overridden with role="presentation"

  Element's default semantics were not overridden with role="none"

  Element has no title attribute or the title attribute is empty

 

Note:

Title also not defined for ‘horizontal and vertical’ icon controls.

State property also not defined as selected or not for ‘horizontal and vertical’ icon controls.

 

MAS Reference:

MAS 4.1.2 - Name, Role, Value

 

Reference Links

Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.
External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information please use this Link. “HCL Staff should not log any third party external bugs. Should be routed to edad team”.
Please reach out to C&AI Teams channel for any process related queries.
  

![image](https://user-images.githubusercontent.com/2320302/79026010-5440ab80-7b3c-11ea-8248-ed284c11d137.png)
